### DISCUSSION TITLE
Upgrade Fails version 4.1.2 > 4.1.7

### DISCUSSION THREAD

**Discussion by gnossosp** [openstore-ecommerce/OpenStore#209] [Q&A] [answered]
DNN Version: 9.7.2  Open Store version 4.1.2
Installing upgrade 4.1.7 fails. 

After first failure I manually replaced the dll files in the installation package. Restarted app pool and DNN. 

After failure the extension list shows version 4.1.6 for all except the OS_ProductView shows 4.1.2

Failure report:
Failure ExceptionSystem.Exception: Error on 'G:\redacted\Install\Temp\pembelbv\NBrightBuy.dll': See InnerException for details. ---> System.ComponentModel.Win32Exception: The process cannot access the file because it is being used by another process --- End of inner exception stack trace --- at SchwabenCode.QuickIO.Internal.InternalQuickIOCommon.NativeExceptionMapping(String path, Int32 errorCode) at DotNetNuke.Common.Utilities.FileSystemUtils.CopyFile(String sourceFileName, String destFileName) at DotNetNuke.Services.Installer.Util.<>c__DisplayClass172_0.<CopyFile>b__0() at DotNetNuke.Common.Utilities.Internal.RetryableAction.TryIt() at DotNetNuke.Common.Utilities.Internal.RetryableAction.RetryEverySecondFor30Seconds(Action action, String description) at DotNetNuke.Services.Installer.Util.CopyFile(InstallFile installFile, String basePath, Logger log) at DotNetNuke.Services.Installer.Installers.FileInstaller.InstallFile(InstallFile insFile)

Failure Installation Failed - Assembly

Info Rolling back component install - Script

Info Restored backup of previous version - 04.01.03.SqlDataProvider

Info Restored backup of previous version - 04.01.06.SqlDataProvider

Info Restored backup of previous version - Uninstall.SqlDataProvider

Info Component rolled back successfully - Script

Info Rolling back component install - ResourceFile

Failure Exception - Could not find file 'G:\redacted\Install\Temp\pembelbvResource.zip'.

Info Component rolled back successfully - ResourceFile

Info Installation Failed - OS_ProductView

Info Starting Installation - OS_DisplaySearch

Info Starting Installation - Module

Info Module registered successfully - OS_DisplaySearch

Info Component installed successfully - Module

Info Starting Installation - File

Info Component installed successfully - File

Failure Package Installation aborted

Info Installation Failed - OS_DisplaySearch

Info Starting Installation - OS_AddressAdmin

Info Starting Installation - Module

Info Module registered successfully - OS_AddressAdmin

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_AddressAdmin

Info Starting Installation - OS_Payment

Info Starting Installation - Module

Info Module registered successfully - OS_Payment

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_Payment

Info Starting Installation - OS_OrderAdmin

Info Starting Installation - Module

Info Module registered successfully - OS_OrderAdmin

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_OrderAdmin

Info Starting Installation - OS_ProfileForm

Info Starting Installation - Module

Info Module registered successfully - OS_ProfileForm

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_ProfileForm

Info Starting Installation - OS_Checkout

Info Starting Installation - Module

Info Module registered successfully - OS_Checkout

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_Checkout

Info Starting Installation - OS_MiniCart

Info Starting Installation - Module

Info Module registered successfully - OS_MiniCart

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_MiniCart

Info Starting Installation - OS_FullCart

Info Starting Installation - Module

Info Module registered successfully - OS_FullCart

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_FullCart

Info Starting Installation - OS_CategoryMenu

Info Starting Installation - Module

Info Module registered successfully - OS_CategoryMenu

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_CategoryMenu

Info Starting Installation - OS_ProductFilter

Info Starting Installation - Module

Info Module registered successfully - OS_ProductFilter

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_ProductFilter

Info Starting Installation - OS_ItemList

Info Starting Installation - Module

Info Module registered successfully - OS_ItemList

Info Component installed successfully - Module

Failure Package Installation aborted

Info Installation Failed - OS_ItemList

Info Deleted temporary install folder

EndJob Installation Failed

**Comment by DNNMonster** [comment]
Hmmm...  Not sure.  Haven't seen it before. Can you install it again and "repair" the installation or does it error out?

**Reply by gnossosp** [reply]
I did that and same result. The weird thing is the older extension version shown for one module only?

**Comment by DNNMonster** [comment]
@leedavi got any ideas?

**Comment by leedavi** [comment]
I think the old version 4.1.2 is because it's bombing out just before upgrade.  I'm surprised about the v4.1.6 on the modules, but that could be an error in the DNN manifesto.

Did you use the Upgrade package?  "OpenStore_4.1.7_Upgrade.zip"

It looks like it's trying to install "OS_ItemList" 2 times??  So that is strange.

I know of this package being installed successfully, so maybe a lock issue or a DNN issue,

Can you try and load the pre-release: https://github.com/openstore-ecommerce/OpenStore/releases/tag/v4.1.8

Again, make sure you use the upgrade package.  (Although that will only cause a failure, due to the skin not installing.)

**Reply by gnossosp** [reply]
Yes OpenStore_4.1.7_Upgrade.zip...

I broke rule number 27 by trying to upgrade my DNN instance before fixing this AND... that didn't work.

After I get that sorted, I will return to this. Thanks for your help.

**Comment by leedavi** [comment]
OK,  I've had a thought.  
Maybe this is linked to the scheduler being triggered during install.
I wonder if we should suggest turning off the scheduler during the upgrade process.

**Comment by gnossosp** [comment]
It still does not upgrade or repair using the package. I did disable the scheduler and I also  updated site (to 9.10.02) but it seems to be working fine. not sure if these are the correct versions of each file. If I have any other issues I can try to follow install content in the .dnn file.
![NBstoreDll](https://user-images.githubusercontent.com/72934916/140993820-4f145bbf-1c25-4708-8e41-aa34b0aca6d6.jpg)
.

**Reply by leedavi** [reply]
I have just tested the GitHUb download and it works on my install without any issue.

Maybe it's the setup or the version of DNN.

Do you get any messages on the installed report ?

![image](https://user-images.githubusercontent.com/5583141/141076480-775aace2-e7d0-4fa6-aff8-0808f0c0b52c.png)

**Comment by gnossosp** [comment, marked as answer]
I was able to successfully effect the repair of the upgrade to 4.17 but renaming all of the NBrightBuy.*.DLL files (except the NBright.Core.Dll and the NBrightDNN.dll) To NBrightBuy.*.dll.BAK first. The job completed as designed and the files were all replaced. 

Earlier I mentioned attempts to upgrade the version of DNN versions (4.10.02) and ran into "Access Denied" issues with some of the files in the /bin directory.  I used this same technique to solve most of those issues.

![nbright_sucess](https://user-images.githubusercontent.com/72934916/141132866-b32b2e55-03ab-4700-a0d1-5e5b6d3aef8c.jpg)

**Comment by leedavi** [comment]
OK sounds like some security issue on the server.   Could be file locked by the AppPool and hence failing to overwrite, or lack of permissions on the server folder.

Thx for posting the solution.

**Comment by DNNMonster** [comment]
For the record the v4.1.7 release had a small error and the manifest versions were not updated.  This doesn't explain the inability to overwrite your bin dir files but it makes sense of the 4.1.6 tag.
